worlds/jakanddaxter/Rules.py

```diff
@@ -0,0 +1,222 @@
+import typing
+from BaseClasses import CollectionState
+from Options import OptionError
+from . import JakAndDaxterWorld
+from .Options import (EnableOrbsanity,
+                      GlobalOrbsanityBundleSize,
+                      PerLevelOrbsanityBundleSize,
+                      FireCanyonCellCount,
+                      MountainPassCellCount,
+                      LavaTubeCellCount,
+                      CitizenOrbTradeAmount,
+                      OracleOrbTradeAmount)
+from .locs import CellLocations as Cells
+from .Locations import location_table
+from .Levels import level_table
+
+
+def set_orb_trade_rule(world: JakAndDaxterWorld):
+    options = world.options
+    player = world.player
+
+    if options.enable_orbsanity == EnableOrbsanity.option_off:
+        world.can_trade = lambda state, required_orbs, required_previous_trade: (
+            can_trade_vanilla(state, player, world, required_orbs, required_previous_trade))
+    else:
+        world.can_trade = lambda state, required_orbs, required_previous_trade: (
+            can_trade_orbsanity(state, player, world, required_orbs, required_previous_trade))
+
+
+def recalculate_reachable_orbs(state: CollectionState, player: int, world: JakAndDaxterWorld) -> None:
+
+    if not state.prog_items[player]["Reachable Orbs Fresh"]:
+
+        # Recalculate every level, every time the cache is stale, because you don't know
+        # when a specific bundle of orbs in one level may unlock access to another.
+        for level in level_table:
+            state.prog_items[player][f"{level} Reachable Orbs".strip()] = (
```

Comment:
Hi it's me the nitpicking core dev, this could be .lstrip() for a tiny performance boost

(But also, is it *that bad* if the global one is `" Reachable Orbs"`? This value is never visible to the player, so you could just accept the slight internal ugliness for a tiny performance boost)

Reply:
I'd prefer not to leave the hanging chad if I can help it. Future developers will be left thinking "oh, a typo! Let me fix that," when they see the following:

```py
        # Also recalculate the global count, still used even when Orbsanity is Off.
        state.prog_items[player][" Reachable Orbs"] = count_reachable_orbs_global(state, world)
        state.prog_items[player]["Reachable Orbs Fresh"] = True
```

Reply:
Fair

Reply:
I did implement the lstrip though lol